core/vm/instructions.go

```diff
@@ -984,13 +984,13 @@ func makePush(size uint64, pushByteSize int) executionFunc {
 			start   = min(codeLen, int(*pc+1))
 			end     = min(codeLen, start+pushByteSize)
 		)
-		scope.Stack.push(new(uint256.Int).SetBytes(
-			common.RightPadBytes(
-				scope.Contract.Code[start:end],
-				pushByteSize,
-			)),
-		)
+		a := new(uint256.Int).SetBytes(scope.Contract.Code[start:end])
 
+		// Missing bytes: pushByteSize - len(pushData)
+		if missing := pushByteSize - (end - start); missing > 0 {
```

Comment:
Even if we could remove this `if` from the hot-loop and adjust the tracers correspondingly, it's (understandably) a micro optimization that doesn't bring any performance benefits:
```
                                       │ bench_if_removed.txt │      bench_if_not_removed.txt      │                                                                                                                
                                       │        sec/op        │   sec/op     vs base               │                                                                                                                
SimpleLoop/staticcall-identity-100M-12            162.9m ± 0%   160.7m ± 2%  -1.35% (p=0.009 n=10)                                                                                                                  
SimpleLoop/call-identity-100M-12                  183.9m ± 2%   180.8m ± 1%  -1.64% (p=0.001 n=10)
SimpleLoop/loop-100M-12                           182.0m ± 1%   182.0m ± 0%       ~ (p=0.853 n=10)
SimpleLoop/loop2-100M-12                          200.7m ± 0%   208.0m ± 5%  +3.68% (p=0.000 n=10)
SimpleLoop/call-nonexist-100M-12                  496.8m ± 2%   508.8m ± 9%  +2.41% (p=0.011 n=10)
SimpleLoop/call-EOA-100M-12                       142.3m ± 0%   141.1m ± 5%       ~ (p=0.143 n=10)
SimpleLoop/call-reverting-100M-12                 323.7m ± 0%   324.9m ± 1%       ~ (p=0.123 n=10)
geomean                                           220.1m        220.9m       +0.37%
```